lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -321,6 +321,91 @@ def test_index_no_visible_section_in_chapter(self):
         self.assertNotIn('Problem 1', response.content)
         self.assertNotIn('Problem 2', response.content)
 
+    def _create_global_staff_user(self):
+        """
+        Create global staff user and log them in
+        """
+        self.global_staff = GlobalStaffFactory.create()  # pylint: disable=attribute-defined-outside-init
+        self.client.login(username=self.global_staff.username, password='test')
+
+    def _create_url_for_enroll_staff(self):
+        """
+        creates the courseware url and enroll staff url
+        """
+        # create the _next parameter
+        courseware_url = reverse(
+            'courseware_section',
+            kwargs={
+                'course_id': unicode(self.course_key),
+                'chapter': unicode(self.chapter.location.name),
+                'section': unicode(self.section.location.name),
+            }
+        )
+        # create the url for enroll_staff view
+        enroll_url = "{enroll_url}?next={courseware_url}".format(
+            enroll_url=reverse('enroll_staff', kwargs={'course_id': unicode(self.course.id)}),
+            courseware_url=courseware_url
+        )
+        return courseware_url, enroll_url
+
+    @ddt.data(
+        ({'enroll': "Enroll"}, True),
+        ({'dont_enroll': "Don't enroll"}, False))
+    @ddt.unpack
+    def test_enroll_staff_redirection(self, data, enrollment):
+        """
+        Verify unenrolled staff is redirected to correct url.
+        """
+        self._create_global_staff_user()
+        courseware_url, enroll_url = self._create_url_for_enroll_staff()
+        response = self.client.post(enroll_url, data=data, follow=True)
+        self.assertEqual(response.status_code, 200)
+
+        # we were redirected to our current location
+        self.assertIn(302, response.redirect_chain[0])
+        self.assertEqual(len(response.redirect_chain), 1)
+        if enrollment:
+            self.assertRedirects(response, courseware_url)
+        else:
+            self.assertRedirects(response, '/courses/{}/about'.format(unicode(self.course_key)))
+
+    def test_enroll_staff_with_invalid_data(self):
+        """
+        If we try to post with an invalid data pattern, then we'll return a 302
+        """
+        self._create_global_staff_user()
+        __, enroll_url = self._create_url_for_enroll_staff()
+        response = self.client.post(enroll_url, data={'test': "test"})
+        self.assertEqual(response.status_code, 302)  # todo goes to about page.
+
+    def test_courseware_redirection(self):
+        """
```

Comment:
This docstring can be improved. Give a quick overview of what the test is testing for (that the whole workflow works), _then_ give the steps that the test goes through. So something like:

Tests that a global staff member, when accessing a page in a course they're not enrolled in, eventually gets routed through the staff enrollment page and back to the original courseware page they tried to access. The process is as follows:
1) ...